sessionctx/stmtctx/stmtctx.go

```diff
@@ -171,8 +172,8 @@ type StatementContext struct {
 
 	// stmtCache is used to store some statement-related values.
 	stmtCache map[StmtCacheKey]interface{}
-	// resourceGroupTag cache for the current statement resource group tag.
-	resourceGroupTag atomic.Value
+	// resourceGroupTags cache for the current statement resource group tags.
+	resourceGroupTags sync.Map
```

Comment:
Since label is enumerable, I used independent `atomic.Value` to avoid lock contention. [code here](https://github.com/pingcap/tidb/pull/29044/files#diff-76144484b2e2c454e3995f6d0d244cbd4d1f672043462a490b054598b49e7b9aR177)